pkg/network/usm/monitor.go

```diff
@@ -40,6 +41,12 @@ const (
 var (
 	state        = Disabled
 	startupError error
+
+	// knownProtocols maps individual protocol types, to their specification,
+	// for the Monitor to use during its initialisation.
+	knownProtocols = map[protocols.ProtocolType]protocols.ProtocolSpec{
+		protocols.HTTP: http.HttpSpec,
+	}
```

Comment:
shouldn't we add to that map if `http_monitoring_enabled`?

Reply:
The map is meant to holds `ProtocolSpec` objects, not `Protocol` instances. `ProtocolSpecs` are the static definition of a protocol, which are its maps, tail call routers, and factory function (which is the one actually checking the configuration to figure out if monitoring for a specific protocol needs to be enabled).
It is used by `initProtocols` to generically handle the initialisation of the different protocols' monitoring we support. It is also used when a protocol is not enabled to fill the list of `excludedFunctions` used by ebpf-manager to patch out probes & tail calls that won't be used so that they don't risk being rejected by the verifier.